lib/modules/manager/composer/artifacts.ts

```diff
@@ -62,7 +62,12 @@ function getAuthJson(): string | null {
     .findAll({ hostType: PackagistDatasource.id })
     ?.forEach((hostRule) => {
       const { resolvedHost, username, password, token } = hostRule;
-      if (resolvedHost && username && password) {
+      if (resolvedHost === 'api.github.com' && token) {
```

Comment:
```suggestion
      if (resolvedHost === 'github.com' && token) {
```
should only use base domain